stats/series.go

```diff
@@ -0,0 +1,73 @@
+// Stats package includes simple statistics utilities
+package stats
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/constraints"
+)
+
+type Number interface {
+	constraints.Integer | constraints.Float
+}
+
+// Series calculates cumulative statistics on a series of numbers
+// For streamlined use, all operations are designed to be error-free, results
+// may not always be useful though.
+type Series[T Number] struct {
+	count uint
+	min   T
+	max   T
+	total T
+}
+
+func (s *Series[T]) Add(v T) {
+	if s.count <= 0 {
+		s.min = v
+		s.max = v
+		s.total = v
+	} else {
+		if s.min > v {
+			s.min = v
+		}
+		if s.max < v {
+			s.max = v
+		}
+		s.total += v
+	}
+	s.count += 1
+}
+
+func (s Series[T]) Len() uint { return s.count }
+func (s Series[T]) Min() T    { return s.min }
+func (s Series[T]) Max() T    { return s.max }
+func (s Series[T]) Total() T  { return s.total }
+
+func (s Series[T]) Avg() T {
+	if s.count <= 0 {
+		return 0
+	}
+	return s.total / T(s.count)
+}
+
+func (s Series[T]) Format(f fmt.State, verb rune) {
```

Comment:
This function is pretty hard to understand. while the test makes it easier, I would add a doc comment.

Reply:
check now

Reply:
I guess? I worry more about the why than the what, meaning why do we want the formatting to be as it is in `rebuildFmtStr` or what can I expect the formatted string to look like

Reply:
@Omeramsc explaining what is the Formatter interface and how to implement it is IMO outside the scope of the code or the comments on it. 

`rebuildFmtStr` is basically rebuilding the format string that was passed to the `fmt` module function that is currently printing our object, since its only provided to us in its de-constructed form.

Whet this is doing is very simple: When you do `fmt.Printf("some '%' format expression", series_object")` it uses the format expression for formatting the numbers in the series object. This allows me to use everything the `fmt` package knows how to do for the numbers without inventing my own custom API for it.